unit_test/main_options_test.py

```diff
@@ -0,0 +1,202 @@
+import pytest
+
+import sys
+
+from cibuildwheel.__main__ import main
+from cibuildwheel.environment import ParsedEnvironment
+from cibuildwheel.util import BuildSelector
+
+from main_util_fixtures import mock_protection, platform, intercepted_build_args
+
+
+
+# CIBW_PLATFORM is tested in main_platform_test.py
+
+
+def test_output_dir(platform, intercepted_build_args, monkeypatch):
+    OUTPUT_DIR = 'some_output_dir'
+
+    monkeypatch.setenv('CIBW_OUTPUT_DIR', OUTPUT_DIR)
+
+    main()
+
+    assert intercepted_build_args.kwargs['output_dir'] == OUTPUT_DIR
+
+
+def test_output_dir_default(platform, intercepted_build_args, monkeypatch):
+    main()
+
+    assert intercepted_build_args.kwargs['output_dir'] == 'wheelhouse'
+
+
+@pytest.mark.parametrize('also_set_environment', [False, True])
+def test_output_dir_argument(also_set_environment, platform, intercepted_build_args, monkeypatch):
+    OUTPUT_DIR = 'some_output_dir'
+
+    monkeypatch.setattr(sys, 'argv', sys.argv + ['--output-dir', OUTPUT_DIR])
+    if also_set_environment:
+        monkeypatch.setenv('CIBW_OUTPUT_DIR', 'not_this_output_dir')
+
+    main()
+
+    assert intercepted_build_args.kwargs['output_dir'] == OUTPUT_DIR
+
+
+def test_build_selector(platform, intercepted_build_args, monkeypatch):
+    BUILD = 'some build* *-selector'
+    SKIP = 'some skip* *-selector'
+
+    monkeypatch.setenv('CIBW_BUILD', BUILD)
+    monkeypatch.setenv('CIBW_SKIP', SKIP)
+
+    main()
+    
+    intercepted_build_selector = intercepted_build_args.kwargs['build_selector']
+    assert isinstance(intercepted_build_selector, BuildSelector)
+    assert intercepted_build_selector('build-this')
+    assert not intercepted_build_selector('skip-that')
+    # This unit test is just testing the options of 'main'
+    # Unit tests for BuildSelector are in build_selector_test.py
+
+
+@pytest.mark.parametrize('architecture, image, full_image', [
+    ('x86_64', None, 'quay.io/pypa/manylinux2010_x86_64'), 
+    ('x86_64', 'manylinux1', 'quay.io/pypa/manylinux1_x86_64'),
+    ('x86_64', 'manylinux2010', 'quay.io/pypa/manylinux2010_x86_64'),
+    ('x86_64', 'custom_image', None),
+    ('i686', None, 'quay.io/pypa/manylinux2010_i686'), 
+    ('i686', 'manylinux1', 'quay.io/pypa/manylinux1_i686'),
+    ('i686', 'manylinux2010', 'quay.io/pypa/manylinux2010_i686'),
+    ('i686', 'custom_image', None),
+])
+def test_manylinux_images(architecture, image, full_image, platform, intercepted_build_args, monkeypatch):
+    if image is not None:
+        monkeypatch.setenv('CIBW_MANYLINUX_' + architecture.upper() + '_IMAGE', image)
+
+    main()
+
+    if platform == 'linux':
+        assert intercepted_build_args.kwargs['manylinux_images'][architecture] == full_image or image
```

Comment:
Just checking this logic condition... It looks to me like `(intercepted_build_args.kwargs['manylinux_images'][architecture] == full_image) or (image is not None)`, is that the intention? If so, it could be written as an `if` more clearly, I think

Reply:
OK, yes, something went wrong, here. I think the full check should be:

```
if image is not None:
    assert intercepted_build_args.kwargs['manylinux_images'][architecture] == full_image
else:
    assert intercepted_build_args.kwargs['manylinux_images'][architecture] == image
```